src/util.py

```diff
@@ -309,6 +309,9 @@ def set_vector_density(data, lat_density=1, lon_density=1, minDistance=0, otherV
         # Get distance between points that are diagonally adjacent
         diagDifference = math.sqrt(latdifference**2 + londifference**2)
 
+        # Initialize ds
+        ds = data.isel(lat=slice(None, None, None), lon=slice(None, None, None))
```

Comment:
I think in addition to initializing `ds` to fix that bug, a warning should also be thrown when the `minDistance` is shorter than the spatial resolution of the data. That way, users know why their plot isn't changing when they specify a smaller `minDistance`.